packages/caliper-core/lib/worker/rate-control/recordRate.js

```diff
@@ -103,7 +106,8 @@ class RecordRateController extends RateInterface {
         offset = buffer.writeUInt32LE(this.records.length, offset);
 
         for (let i = 0; i < this.records.length; i++) {
-            offset = buffer.writeUInt32LE(this.records[i], offset);
+            const time = this.records[i] !== undefined ? this.records[i] : 0;
```

Comment:
technically these changes are not required because an empty element in an array will still be written out as zero. However I have no problem leaving it in as it means we are more explicit in declaring what we want the behaviour to be rather than relying on what the writeUInt32LE method does. But you didn't add it for the writing of BE as well. So either we have it in both places, or not in either place